src/generate-query.ts

```diff
@@ -508,10 +505,56 @@ function getMissingSlicingArg(
   )
   if (usesSlicingArg) return null
 
-  // Return the first slicing arguments:
-  return field.arguments.find((arg) =>
-    slicingArguments.includes(arg.name.value)
+  // Returns the first slicing argument
+  // if slicing argument can be split on '.'
+  // then we have a nested slicing argument,
+  // and we need to follow the path all the way
+  // down to the leaf input field definition
+  return field.arguments.find((arg) => {
+    return slicingArguments.find((slicingArgument) => {
+      return getMissingSlicingArgHelperField(
+        slicingArgument.split('.'),
+        arg,
+        schema
+      )
+    })
+  })
+}
+
+function getMissingSlicingArgHelperFields(
+  tokenizedPath: string[],
+  type: GraphQLNamedType,
+  schema: GraphQLSchema
+): InputValueDefinitionNode {
+  if (type?.astNode?.kind === Kind.INPUT_OBJECT_TYPE_DEFINITION) {
+    return type.astNode.fields.find((field) => {
+      let ret = getMissingSlicingArgHelperField(
+        tokenizedPath.slice(1),
+        field,
+        schema
+      )
+      return ret
+    })
+  }
+}
+
+function getMissingSlicingArgHelperField(
+  tokenizedPath: string[],
+  field: InputValueDefinitionNode,
+  schema: GraphQLSchema
+): InputValueDefinitionNode {
+  if (tokenizedPath.length < 1) return undefined
+  else if (field.name.value !== tokenizedPath[0]) return undefined
+  else if (tokenizedPath.length === 1) return field
+
+  if (
+    field.kind !== Kind.INPUT_VALUE_DEFINITION ||
+    field.type.kind !== Kind.NAMED_TYPE
```

Comment:
If the type isn't a named type, then should we try to unwrap it? Or is this not a case?

Reply:
removed function so this is outdated, resolving convo